code/modules/events/crystal_event.dm

```diff
@@ -256,6 +257,33 @@ This section is for the destabilized SM
 		active = FALSE
 		is_stabilized = TRUE
 
+/obj/machinery/destabilized_crystal/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The Crystal appears to be heavily destabilized. Maybe it can be fixed by injecting it with something from another world.</span>"
+
+/obj/machinery/destabilized_crystal/Bumped(atom/movable/AM)
+	if(isliving(AM))
+		AM.visible_message("<span class='danger'>\The [AM] slams into \the [src] inducing a resonance... [AM.p_their()] body starts to glow and burst into flames before flashing into dust!</span>",\
+		"<span class='userdanger'>You slam into \the [src] as your ears are filled with unearthly ringing. Your last thought is \"Oh, fuck.\"</span>",\
+		"<span class='hear'>You hear an unearthly noise as a wave of heat washes over you.</span>")
+	else
+		return
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, TRUE)
+	Consume(AM)
+
+/obj/machinery/destabilized_crystal/proc/Consume(atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/user = AM
+		if(user.status_flags & GODMODE)
+			return
+		message_admins("[src] has consumed [key_name_admin(user)] [ADMIN_JMP(src)].")
+		investigate_log("has consumed [key_name(user)].", INVESTIGATE_SUPERMATTER)
+		user.dust(force = TRUE)
+	for(var/obj/structure/crystal_portal/portals in GLOB.crystal_portals)
```

Comment:
Can `crystal_portals` contain anything other than `/obj/structure/crystal_portal`? If not, then you should make this an untyped loop, then type it inside the body.

Also, if `portals` is only one portal, it shouldn't be plural.

Reply:
Will do the typing inside the body since it can only contain the portals and I'll change the var name to singular